chromadb/proto/convert.py

```diff
@@ -19,42 +19,43 @@
     VectorEmbeddingRecord,
     VectorQueryResult,
 )
+import numpy as np
+from numpy.typing import NDArray
 
 
 # TODO: Unit tests for this file, handling optional states etc
 def to_proto_vector(vector: Vector, encoding: ScalarEncoding) -> proto.Vector:
     if encoding == ScalarEncoding.FLOAT32:
-        as_bytes = array.array("f", vector).tobytes()
+        as_bytes = np.array(vector, dtype=np.float32).tobytes()
         proto_encoding = proto.ScalarEncoding.FLOAT32
     elif encoding == ScalarEncoding.INT32:
-        as_bytes = array.array("i", vector).tobytes()
+        as_bytes = np.array(vector, dtype=np.int32).tobytes()
         proto_encoding = proto.ScalarEncoding.INT32
     else:
         raise ValueError(
             f"Unknown encoding {encoding}, expected one of {ScalarEncoding.FLOAT32} \
             or {ScalarEncoding.INT32}"
         )
 
-    return proto.Vector(dimension=len(vector), vector=as_bytes, encoding=proto_encoding)
+    return proto.Vector(dimension=vector.size, vector=as_bytes, encoding=proto_encoding)
 
 
 def from_proto_vector(vector: proto.Vector) -> Tuple[Embedding, ScalarEncoding]:
     encoding = vector.encoding
-    as_array: Union[array.array[float], array.array[int]]
+    as_array: Union[NDArray[np.int32], NDArray[np.float32]]
     if encoding == proto.ScalarEncoding.FLOAT32:
-        as_array = array.array("f")
+        as_array = np.frombuffer(vector.vector, dtype=np.float32)
```

Comment:
Have we reasoned through any incompatabilities between the python byte representation and numpys

Reply:
I've tested this and consulted the internet, and it seems safe. However, the best way to test is to actually run it through the logservice in our tests, which I can work on.